spec/advanced/binarySearchTree_test.js

```diff
@@ -14,9 +14,56 @@ describe('BinarySearchTree', () => {
 		it('returns the nodes data', () => {
 			const bst = new BinarySearchTree()
 			bst.insert(3)
-			console.log(bst);
+			bst.insert(9)
+			bst.insert(15)
 
-			expect(leastNode.getData() ).to.deep.equal({data: 3})
+			expect(bst.count).to.equal(3)
+		})
+	})
+
+	context('search(value)', () => {
+		it('returns the nodes data', () => {
+			const bst = new BinarySearchTree()
+			bst.insert(3)
+			bst.insert(9)
+			bst.insert(15)
+
+			expect(bst.search(3) ).to.equal(3)
+			expect(bst.search(7) ).to.equal(false)
+		})
+	})
+
+	context('remove(value)', () => {
+		it('removes a values node (if exists) from the tree', () => {
+			const bst = new BinarySearchTree()
+			bst.insert(3)
+			bst.insert(9)
+			bst.insert(15)
+
+			bst.remove(3)
+			expect(bst.count).to.equal(2)
+		})
+	})
+
+	context('traverse(callback)', () => {
+		it('traverse the tree using in-order traversal and apply function on each nodes value', () => {
+			const bst = new BinarySearchTree()
+			bst.insert(3)
+			bst.insert(9)
+			bst.insert(15)
+
+			bst.traverse((val) => console.log(val))
+		})
+	})
+
+	context('count()', () => {
+		it('return the number of nodes in the tree', () => {
+			const bst = new BinarySearchTree()
+			bst.insert(3)
+			bst.insert(9)
+			bst.insert(15)
+
+			expect(bst.count).to.equal(3)
 		})
 	})
```

Comment:
The tests look good, I don't think you have full coverage however. Not a spec in this goal so I wouldn't trip but if you wanted, you could install the npm module nyc-istanbul, and run your test with nyc it will tell you what lines are not being tested. " nyc mocha --test stuff "